system-helper/flatpak-system-helper.c

```diff
@@ -1553,11 +2183,15 @@ main (int    argc,
                                   NULL,
                                   NULL);
 
+  ongoing_pulls = g_ptr_array_new ();
+
   /* Ensure we don't idle exit */
   schedule_idle_callback ();
 
   main_loop = g_main_loop_new (NULL, FALSE);
   g_main_loop_run (main_loop);
 
+  g_clear_pointer (&ongoing_pulls, g_ptr_array_unref);
```

Comment:
I guess it doesn't matter much, but: does this leak remaining pulls ? Is ongoing_pulls guaranteed empty at this point ?

Reply:
Yes, it should be empty at this point because we block the idle handler from exiting. Check 
`idle_timeout_cb`

Reply:
ah right